### PR TITLE
[backend] Add Base Image Collections

### PR DESCRIPTION
**Pull request by rbino**
This PR adds a CRUD for Base Image Collections, which is the starting point to support Base Images and OTA Campaigns.

### CODE REVIEW

**Comment by szakhlypa** [comment]
You may also update `seeds.exs` to create a new `base_image_collection`.

File: backend/test/edgehog/base_images_test.exs [hunk 1]
```diff
@@ -27,48 +27,68 @@ defmodule Edgehog.BaseImagesTest do
...
+    } do
+      base_image_collection = base_image_collection_fixture(system_model)
 
       assert BaseImages.get_base_image_collection!(base_image_collection.id) ==
                base_image_collection
     end
 
-    test "create_base_image_collection/1 with valid data creates a base_image_collection" do
-      valid_attrs = %{handle: "some handle", name: "some name"}
+    test "create_base_image_collection/1 with valid data creates a base_image_collection", %{
```
Comment (davidebriani) [inline backend/test/edgehog/base_images_test.exs:56]: You should update the signature from `create_base_image_collection/1` to `create_base_image_collection/2`.

File: backend/lib/edgehog/base_images.ex [hunk 2]
```diff
@@ -69,8 +70,8 @@ defmodule Edgehog.BaseImages do
       {:error, %Ecto.Changeset{}}
 
   """
-  def create_base_image_collection(attrs \\ %{}) do
-    %BaseImageCollection{}
+  def create_base_image_collection(%Devices.SystemModel{} = system_model, attrs \\ %{}) do
+    %BaseImageCollection{system_model_id: system_model.id}
```
Comment (davidebriani) [inline backend/lib/edgehog/base_images.ex:82]: You should update the documentation example above to respect the updated signature, requiring a system model as the first argument.

File: backend/lib/edgehog_web/schema/base_images_types.ex [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+  A base image collection represents the collection of all Base Images that \
+  can run on a specific System Model.
+  """
+  node object(:base_image_collection) do
+    @desc "The display name of the base image collection."
+    field :name, non_null(:string)
+
+    @desc "The identifier of the system model."
+    field :handle, non_null(:string)
```
Comment (davidebriani) [inline backend/lib/edgehog_web/schema/base_images_types.ex:38]: The identifier of the ~~system model~~ **base image collection**

**Review by davidebriani — CHANGES_REQUESTED** [review]

**Review by davidebriani — APPROVED** [review]

File: backend/test/edgehog/base_images_test.exs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  describe "base_image_collections" do
+    alias Edgehog.BaseImages.BaseImageCollection
+
+    import Edgehog.BaseImagesFixtures
+    import Edgehog.DevicesFixtures
+
+    setup do
+      hardware_type = hardware_type_fixture()
+
+      {:ok, hardware_type: hardware_type, system_model: system_model_fixture(hardware_type)}
```
Comment (szakhlypa) [inline backend/test/edgehog/base_images_test.exs:35]: There is no usage of `hardware_type` in tests, so we may not add it to the metadata.

File: backend/test/edgehog_web/schema/mutation/create_base_image_collection_test.exs [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  alias Edgehog.BaseImages
+  alias Edgehog.BaseImages.BaseImageCollection
+
+  import Edgehog.DevicesFixtures
+
+  describe "createBaseImageCollection field" do
+    setup do
+      hardware_type = hardware_type_fixture()
+
+      {:ok, hardware_type: hardware_type, system_model: system_model_fixture(hardware_type)}
```
Comment (szakhlypa) [inline backend/test/edgehog_web/schema/mutation/create_base_image_collection_test.exs:33]: There is no usage of hardware_type in tests, so we may not add it to the metadata.

**Review by szakhlypa — APPROVED** [review]

**Review by szakhlypa — APPROVED** [review]

**Pull request merged**